### PR TITLE
Restrict file access

### LINKED ISSUE #146: Restrict file access
We should restrict the access to the files generated by the wallet. 

Today they are created with 0666 permission, we should change it to 0600. Also, we should verify the user hasn't changed this permission afterward.

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #146

### CODE REVIEW

File: service/store/v1/store.go [hunk 1]
```diff
@@ -169,14 +163,36 @@ func (s *Store) removeExistingRSAKeys() error {
...
+		return nil, err
+	}
+
+	stats, err := os.Stat(fileName)
+	if err != nil {
+		return nil, err
+	}
+	if stats.Mode().Perm() != fs.FileMode(0600) {
+		return nil, fmt.Errorf("file \"%s\" has not the right permission, expected 0600", fileName)
+	}
```
Comment (jeremyletang) [inline service/store/v1/store.go:178]: Not sure that check is correct. can we say `>=`. Validating against insufficient permission should be an error, but if the file have more permission that expected, the program should probably not fail?

File: service/store/v1/store.go [hunk 2]
```diff
@@ -169,14 +163,36 @@ func (s *Store) removeExistingRSAKeys() error {
 	return createFolder(s.keyFolderPath)
 }
 
+func readFile(fileName string) ([]byte, error) {
+	content, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		return nil, err
```
Comment (jeremyletang) [inline service/store/v1/store.go:169]: As of go 1.16 ioutil should be avoided, and features from this package should be replaced by the io and os pacakge.

What about introducing that in here while doing the changes on files restriction? could be a good time to migrate to the new fs.FS api too: https://pkg.go.dev/io/fs#FS.

What do you thing?

File: service/store/v1/store.go (hunk 1 above)
Comment (ValentinTrinque) [reply service/store/v1/store.go:178]: About not enough permissions, I think the read and write are going to fail anyway, so I would say that's already ok.

> If the file have more permission that expected, the program should probably not fail.

This is measure similar to the one of SSH has put in place, so we can ensure sensitive files are not open to the world.

What do you think about this ?

File: service/store/v1/store.go (hunk 2 above)
Comment (ValentinTrinque) [reply service/store/v1/store.go:169]: You are right

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**